client/cmd.go

```diff
@@ -104,9 +104,10 @@ func ReadPersistentCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Cont
 
 	if clientCtx.Keyring == nil || flagSet.Changed(flags.FlagKeyringBackend) {
 		keyringBackend, _ := flagSet.GetString(flags.FlagKeyringBackend)
+		keyringDir, _ := flags.GetKeyringDir(flagSet, clientCtx.HomeDir)
```

Comment:
This has to be a new flag. `--home` and `--keyring-dir` can be used together.

Reply:
Yeah there should be a way to initialise the keyring with a different home directory (when setting a home directory makes sense at all, i.e. `keyring-backend = file | test | pass`)

Reply:
> This has to be a new flag. `--home` and `--keyring-dir` can be used together.

I don't understand what your critique is.  `flags.GetKeyringDir` uses `--keyring-dir` (which can be different from `--home`), and it defaults to the second argument, and missing that, the value of `--home`.

Did I not implement that correctly?

Reply:
No, sorry, I misread this code. But I still find this confusing. See comment on `GetKeyringDir`.